napari/_qt/widgets/qt_viewer_dock_widget.py

```diff
@@ -258,13 +267,33 @@ def __init__(self, parent, title: str = '', vertical=False):
         line = QFrame(self)
         line.setObjectName("QtCustomTitleBarLine")
 
-        self.close_button = QPushButton(self)
-        self.close_button.setToolTip(trans._('hide this panel'))
-        self.close_button.setObjectName("QTitleBarCloseButton")
-        self.close_button.setCursor(Qt.ArrowCursor)
-        self.close_button.clicked.connect(
-            lambda: self.parent().toggleViewAction().trigger()
-        )
+        add_close = False
+        try:
+            # if the plugins menu is already created, check to see if this is a plugin
+            # dock widget.  If it is, then add the close button option to the title bar.
+            actions = [
+                action.text()
+                for action in self.parent().qt_viewer.viewer.window.plugins_menu.actions()
+            ]
+            if self.parent().name in actions:
+                add_close = True
+                self.close_button = QPushButton(self)
+                self.close_button.setToolTip(trans._('close this panel'))
+                self.close_button.setObjectName("QTitleBarCloseButton")
+                self.close_button.setCursor(Qt.ArrowCursor)
+                self.close_button.clicked.connect(
+                    lambda: self.parent().destroyOnClose()
+                )
+            else:
+                add_close = False
+        except AttributeError:
```

Comment:
what causes the attribute error here?

Reply:
This will fail on start up.  I think `QtCustomTitleBar` is called for the layer list and layer controls before the plugins menu is created.